learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+
+`apiKey` must be a valid Meilisearch API key. A tenant token will have access to the same indexes as the API key you use when generating it. If no API key is provided, the SDK might be able to infer it automatically.
+
+`expiryDate` is optional when using an SDK. If used, this parameter must be a UNIX timestamp. Tokens becomes invalid after their `expiryDate`.
+
+You can read more about each of element of the tenant token payload in [this guide's final section](#tenant-token-payload-reference).
+
+### Using a tenant token with an SDK
+
+The resulting token, stored on the `token` variable, can then be sent back to the front-end and used to make queries that will only return results whose `user_id` attribute equals the current user's ID:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_sdk_search_1">
+const frontEndClient = new Meilisearch({
+  apiKey: token
+});
+
+const medicalRecordsIndex = frontEndClient.index('patient_medical_records');
+const search = medicalRecordsIndex.search('blood test');
+```
+
+Applications may use tenant tokens and API keys interchangeably when searching. For example, the same application might use a default search API key for queries on public indexes and a tenant token for logged-in users searching on private data.
+
+## Generating tenant tokens without a Meilisearch SDK
+
+Though [Meilisearch recommends using an official SDK](#generating-tenant-tokens-with-an-sdk), this is not a requirement. Since tenant tokens follow the [JWT standard](https://jwt.io), you can use a number of [compatible third-party helper libraries](https://jwt.io/libraries). You may also skip all assistance and generate a token from scratch.
+
+The full process requires you to [create a token header](#step-1-creating-a-token-header), [prepare a data payload](#step-2-preparing-the-token-payload) with at least one set of search rules, and then [sign the token](#step-3-signing-a-token) with an API key.
+
+If you are already familiar with the creation of JWTs and only want to know about the specific requirements of a tenant token payload, take a look at this guide's [token payload reference](#tenant-token-payload-reference).
+
+::: note
+The following examples use JavaScript for the sake of simplicity, but Meilisearch should be compatible with the majority of modern stacks and languages.
+:::
+
+### Step 1: Creating a token header
+
+The first step when generating a tenant token is to create a header stating token type and the chosen encryption algorithm. Once that is done you must encode the header object into `base64`:
+
+```js
+let header = {
+  "alg": "HS256",
+  "typ": "JWT"
+}
+
+const base64Header = base64Encode(header);
```

Comment:
This function does not exists, the user has to use the `Buffer` API. 

```js
function encode64(str: any) {
  return Buffer.from(JSON.stringify(str)).toString('base64')
}
```

Reply:
Thanks, @bidoubiwa! I think this is the pseudo-code used in the specification that was presented here. I will update the specification when the js method is validated by someone from the @meilisearch/integration-team. 💪

Reply:
Here you go: https://github.com/meilisearch/meilisearch-js/blob/e7b9c83ad6feb6044c76c8d71bf51372b2dbf34f/src/lib/token.ts